core/storage/config/gae_models.py

```diff
@@ -61,3 +61,54 @@ def get_export_policy():
 
     def commit(self, committer_id, commit_cmds):
         super(ConfigPropertyModel, self).commit(committer_id, '', commit_cmds)
+
+
+class PlatformParameterSnapshotMetadataModel(
+        base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a platform parameter snapshot."""
+
+    pass
+
+
+class PlatformParameterSnapshotContentModel(
+        base_models.BaseSnapshotContentModel):
+    """Storage model for the content for a platform parameter snapshot."""
+
+    pass
+
+
+class PlatformParameterModel(base_models.VersionedModel):
+    """A class that represents a named dynamic platform parameter.
+    This model only stores fields that can be updated in run time.
+
+    The id is the name of the parameter.
+    """
+
+    SNAPSHOT_METADATA_CLASS = PlatformParameterSnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = PlatformParameterSnapshotContentModel
+
+    rules = ndb.JsonProperty(repeated=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """PlatformParameterModel is not related to users."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_export_policy():
+        """Model does not contain user data."""
+        return base_models.EXPORT_POLICY.NOT_APPLICABLE
+
+    @classmethod
+    def create(cls, param_name, rule_dicts):
+        """Creates a PlatformParameterModel instance.
+
+        Args:
+            param_name: str. The context for evaluation.
+            rule_dicts: list(dict). List of dict mappings of rule objects.
```

Comment:
Not enough detail. Explain the internal structure of each dict. 

Also we should be careful about the dict format changing in the future. I don't see a schema version field, though? /cc @BenHenning

Reply:
Done, added structure details of the dict.
Also added a rule_schema_version field in the model, I think it's enough for identifying the schema version of stored models if we need to update schema, is that right?